### ISSUE TITLE
Symbol's function definition is void: omnisharp--valid-solution-path-p

### ISSUE THREAD

**Issue by tushartyagi** [OmniSharp/omnisharp-emacs#253]
Hi,

I am trying to run omnisharp-emacs with the following configuration:
omnisharp-20160920.2343
debian 8

So far I have tried both omnisharp-server and omnisharp-roslyn with the above configuration but I am not able to run my projects.
Starting the server in the project directory works for both servers, but I am not able to access it using emacs.

For both of the servers, using `omnisharp-start-omnisharp-server` gives the following error message:
`omnisharp-start-omnisharp-server: Symbol's function definition is void: omnisharp--valid-solution-path-p`

Trying to manually run `omnisharp-mode` in the CS files gives the following error in _Messages_ buffer:

`Eager macro-expansion failure: (wrong-number-of-arguments (2 . 3) 4) [2 times] 
File mode specification error: (wrong-number-of-arguments (2 . 3) 4)`

I've followed the exact same steps as listed in the wiki to run the server. 

The strange thing is that this is working perfectly fine on mac with the exact same config (but the intellisense is slow beyond use, that is a different case).

Any ideas?
Or how can I use the server which starts in the project directory?

**Comment by jqtruong** [comment]
you could add 
```emacs-lisp
(require 'omnisharp-settings)
(require 'omnisharp-utils)
```
to your `.emacs`, but the main issue is that those two files were commented out from `omnisharp-autoloads`; i just started playing around with it and didn't dig any further as to why they were.

**Comment by razzmatazz** [comment]
@jqtruong are you using the "feature-roslyn-support" branch?

Because the version on the "master" branc & ELPA release does not support the `omnisharp-roslyn` server and is about to be deprecated.

**Comment by bojanrajkovic** [comment]
I have a similar issue on Windows, using the latest release (via Spacemacs). I keep getting this every time I try to start the server:

```
omnisharp: Starting OmniSharpServer using project folder/solution file: c:/VCS/Oxide
omnisharp: using the server at: C:\omnisharp-win-x64-net46\OmniSharp.exe
omnisharp--do-server-start: Symbol's function definition is void: omnisharp--path-to-server
```
`eval`-ing a buffer with a call to the function in it succeeds, but I can't get the server to be started correctly no matter what I try.

**Comment by razzmatazz** [comment]
@bojanrajkovic I think this is a separate issue for you.. you have `system-type` = "cygwin" but for some reason you might not have these functions defined: https://www.gnu.org/software/emacs/manual/html_node/elisp/File-Names.html#DOCF13

or am I wrong?

**Comment by bojanrajkovic** [comment]
I'm not using Cygwin, so I'm not sure why that would be my System type. I'll check when I get back to my machine what the system type is being reported as.

________________________________

**Comment by razzmatazz** [comment]
Ok. Then maybe I was wrong in my presumptions.
Bojan Rajkovic <notifications@github.com> schrieb am Di. 11. Apr. 2017 um
07:23:

**Comment by pantaaron** [comment]
I am getting the same issue as @bojanrajkovic, and I am on Mac with Emacs Client for OS X. I have tested this with both 25.1.1 and 25.2.1. My .emacs contains `(setq omnisharp-server-executable-path "/usr/local/bin/omnisharp")` and I have even tried `(setq omnisharp-server-executable-path '("/usr/local/bin/omnisharp"))`. The server exists in my directory and runs properly in terminal.

**Comment by bojanrajkovic** [comment]
@pantaaron's solution did it for me as well--I was under the impression that the spacemacs C# layer would do this for me, but it didn't. Maybe the layer needs fixing. :)

**Comment by razzmatazz** [comment]
I have a suspicion this (function definition is void) problem has been fixed by https://github.com/OmniSharp/omnisharp-emacs/pull/314

Could you check if it has been fixed by the latest version?

Also, do you have company mode enabled as part of `csharp-mode-hook? What does `M-x company-complete` do for you?

**Comment by razzmatazz** [comment]
Assuming the underlying problem is fixed by #314, please reopen if this is not true.
